app/monitoring/CloudWatch.scala

```diff
@@ -0,0 +1,65 @@
+package monitoring
+
+import java.util.concurrent.Future
+
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.regions.Regions.EU_WEST_1
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchAsyncClient
+import com.amazonaws.services.cloudwatch.model.{Dimension, MetricDatum, PutMetricDataRequest, PutMetricDataResult}
+import com.gu.aws.CredentialsProvider
+import monitoring.CloudWatch.cloudwatch
+
+trait CloudWatch extends TagAwareLogger {
+  val stage: String
+  val application : String
+  lazy val stageDimension: Dimension = new Dimension().withName("Stage").withValue(stage)
+  def mandatoryDimensions:Seq[Dimension] = Seq(stageDimension)
+
+  trait LoggingAsyncHandler extends AsyncHandler[PutMetricDataRequest, PutMetricDataResult]
```

Comment:
could we keep the styling consistent with the rest of the code base?  e.g.1. `trait Foo {` rather than having `{` on a new line; e.g.2 being explicit on the return type of functions that return a unit (`def foobar(): Unit = ...`)